packages/core/src/http/response/http.responseHeaders.factory.ts

```diff
@@ -10,8 +22,22 @@ export const headersFactory = (data: {
   body: any;
   path: string;
   status: HttpStatus;
-}) => (headers?: HttpHeaders) => ({
-  ...DEFAULT_HEADERS,
-  ...contentTypeFactory(data),
-  ...headers,
-});
+}) => (headers?: HttpHeaders) => {
+  const mergedHeaders = {
+    ...DEFAULT_HEADERS,
+    ...contentTypeFactory(data),
+    ...headers
+  };
+
+  if (!useLowerCaseHeadersOnly()) {
+    return mergedHeaders;
+  }
+
+  // lower case header
+  const lowerCasedHeaders = {};
```

Comment:
I would extract this process to separate function, eg. `#normalizeHeaders` - then you can use simple ternary check. If we care here about the performance impact then it can stay as it is, but if not it would be better to use `reduce` instead of `forEach`.  In typical case the total number of outgoing headers won't reach more than 10.